server/services/publish-message-action.js

```diff
@@ -6,14 +6,30 @@ const scheduleSender = new MessageSender(config.scheduleQueueConfig, config.sche
 
 async function publishMessageAction (claim) {
   try {
+    const calculationMessage = getCalculationMessage(claim)
+    const scheduleMessage = getScheduleMessage(claim)
     await Promise.all([
-      calculationSender.sendMessage(claim),
-      scheduleSender.sendMessage(claim)
+      calculationSender.sendMessage(calculationMessage),
+      scheduleSender.sendMessage(scheduleMessage)
     ])
   } catch (err) {
     console.log(err)
     throw err
   }
 }
 
-module.exports = { publishMessageAction }
+function getCalculationMessage (claim) {
+  return claim
+}
```

Comment:
I appreciate this is improving consistency, but it feels a little like premature optimisation. I can see it's needed for the schedule message, but it seems unnecessary for the calculation message

Reply:
Agreed but I included an explanation of why this is needed for Pact in Confluence.

_A prerequisite of the Pact message provider being able to test from the provider side, is that the application must have a function that outputs the message content so Pact can validate it against a contract.

In the claim service, this was not the case as the message content is received via an Http post and passed directly to SQS.  To support contract testing, the code was rewritten as below.  This does solve some outstanding technical debt where the full claim was unnecessarily sent to the schedule queue when only the claimId was needed but does introduce extra complexity to the calculation queue logic that delivers no application benefit._

Reply:
That sounds good, and making code more testable is never a bad thing.